src/micro_error.c

```diff
@@ -107,7 +107,7 @@ micro_ErrorFromStatus(natsStatus s)
     err = NATS_CALLOC(1, sizeof(microError) + message_len + 1);
     if (err == NULL)
         return &_errorOutOfMemory;
-
+    err->message = (char *)(err + 1);
```

Comment:
As of now, it is declared as `char *message` (at least in this PR). That is actually what I do (single alloc of struct + some content) with `struct __natsMsg`. 

I would declare `const char* message`, to indicate that this should not be freed individually, and add a comment before that field to say that it has to be the last in the struct (in case you add something later). So you should have something like this:
```
char *ptr = NULL;
...
ptr = (char*) (((char*) &(err->message)) + sizeof(err->message));
memcpy(ptr, message, message_len + 1);
err->message = (const char*) ptr;
```

Reply:
@kozlovic I changed to `const`, 👍 it's cleaner. I decided on `ptr=(char*)err + sizeof(*err)` as the least controversial way to calculate `ptr`. Please take a look.